src/Documentation/Advanced-Concepts/Docker-Deployment.md

```diff
@@ -0,0 +1,615 @@
+
+`Dockerfile.debug`:
+
+```dockerfile
+FROM microsoft/dotnet:1.1.2-sdk
+ENV NUGET_XMLDOC_MODE skip
+WORKDIR /vsdbg
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+        unzip \
+    && rm -rf /var/lib/apt/lists/* \
+    && curl -sSL https://aka.ms/getvsdbgsh | bash /dev/stdin -v latest -l /vsdbg 
+WORKDIR /app
+ENTRYPOINT ["tail", "-f", "/dev/null"]
+```
+
+This dockerfile essentially download and install the VSdbg debugger and start an empty container and keep it alive forever so we don't need tear down/up while debugging.
+
+Now, for production, the image is smaller since it contains only the .Net Core runtime and not the whole SDK, and the dockerfile is a bit simpler:
+
+`Dockerfile`:
+
+```dockerfile
+FROM microsoft/dotnet:1.1.2-runtime
+WORKDIR /app
+ENTRYPOINT ["dotnet", "OrleansSilo.dll"]
+COPY . /app
+```
+
+# docker-compose
+
+The `docker-compose.yml` file, essentially pack up within a project a set of services and its dependencies at service level. Each service contains one or more instances of a given container, which is based on the images you selected on your Dockerfile. More details on the `docker-compose` you you can find on [docker-compose documentation](https://docs.docker.com/compose/). 
+
+For an Orleans deployment, a common use case is to have a `docker-compose.yml` which contains two services. One for Orleans Silo, and other for Orleans Client. The Client would have a dependency on the Silo and that means, it will only start after the Silo service is up. Another case is to add a storage/database service/container, like for example SQL Server, which should start first before the client and the silo, so both services should take a dependency on it.
+
+> **Note**: Before you read further (and eventually get crazy with it), please note that _identation_ **matters** in `docker-compose` files. So pay attention to it if you have any problem.
+
+Here is how we will describe our services for this article:
+
+`docker-compose.override.yml` (Debug):
+
+```yml
+version: '3.1'
+
+services:
+  orleans-client:
+    image: orleans-client:debug
+    build:
+      context: ./src/OrleansClient/bin/PublishOutput/
+      dockerfile: Dockerfile.Debug
+    volumes: 
+      - ./src/OrleansClient/bin/PublishOutput/:/app
+      - ~/.nuget/packages:/root/.nuget/packages:ro
+    depends_on: 
+      - orleans-silo
+  orleans-silo:
+    image: orleans-silo:debug
+    build:
+      context: ./src/OrleansSilo/bin/PublishOutput/
+      dockerfile: Dockerfile.Debug
+    volumes: 
+      - ./src/OrleansSilo/bin/PublishOutput/:/app
+      - ~/.nuget/packages:/root/.nuget/packages:ro
+```
+
+`docker-compose.yml` (production):
+```yml
+version: '3.1'
+
+services:
+  orleans-client:
+    image: orleans-client
+    depends_on: 
```

Comment:
Ha! That is new. Didn't checked that update. It used to work... Will check with the Docker folks what should be used now. Thanks for point out.